src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(prey) {
+    if (prey instanceof Herbivore && !prey.hidden) {
+      prey.health -= 50;
+
+      if (prey.health === 0) {
```

Comment:
Do you realy need if() - condition to filter the array?

Reply:
There could be situations when prey.health decreases to 50, not to 0. In this case, filtering is useless. So I decided to add If() - condition to avoid running an unnecessary loop.

Reply:
@VolodymyrChuyko you already know what object you changed. fined it inside the array and delete it from the Animal array without copping it by filter)